committeeoversightapp/views.py

```diff
@@ -135,46 +171,28 @@ def prepare_results(self, qs):
                             kwargs={'pk':item.pk}
                         ))
                     ),
-                ])
-        else:
-            for item in qs:
-                json_data.append([
-                    item.updated_at.strftime("%Y-%m-%d %I:%M%p %Z"),
-                    detail_string.format(
-                        escape(reverse_lazy(
-                            'detail-event',
-                            kwargs={'pk':item.pk}
-                        )),
-                        escape(item.name.title())
-                    ),
-                    item.start_date,
-                    '',
-                    '',
-                ])
+                ]
+                json_data.append(row_data)
+            # if not authenticated, only include display hearing categories 
+            elif not item.category or item.category.name in settings.DISPLAY_CATEGORIES:
+                row_data += ['', '']
+                json_data.append(row_data)
+            else:
+                pass
 
         return json_data
 
 
 class EventDetail(DetailView):
-    model = Event
+    model = HearingEvent
     template_name = "hearing_detail.html"
     context_object_name = 'hearing'
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        #get category context
-        try:
-            context['category'] = HearingCategory.objects.get(
-                event=context['hearing']
-            ).category
-        except ObjectDoesNotExist:
-            context['category'] = None
-
-        context['committees'] = CommitteeOrganization.objects.filter(
-            eventparticipant__event=context['hearing'].id
-        )
-
+        context['category'] = context['hearing'].category
```

Comment:
What's the reason for explicitly setting these in the context, instead of accessing them via your `hearing` object in the template?

Reply:
A vestigial structure. Ty for catching!